src/xcache-consistency-check

```diff
@@ -460,35 +546,87 @@ def find_path():
             xrootd_p = p
             break
 
-    xrootd_config_file = None
+    xrootd_config_file = ""
     if xrootd_p is not None:
         command_chunks = xrootd_p.split()
         for chunk in command_chunks:
             if "/etc" in chunk:
                 xrootd_config_file = chunk
 
-        if xrootd_p is not None:
+        if xrootd_config_file:
             log.info("found xrootd configuration file:"+ xrootd_config_file)
-
-            #Then use the xrootd config file to find the 'oss.localroot' parameter
-            out = subprocess.Popen(['cconfig', '-c', xrootd_config_file], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-            stdout, stderr = out.communicate()
-            if stderr is None:
-                log.debug("Parsing cconfig output to get 'oss.localroot'")
-            else:
-                log.error("Something went wrong when executing cconfig -c "+ xrootd_config_file)
-
-            for line in stdout.split('\n'):
-                if "oss.localroot" in line:
-                    if len(line.split()) > 0:
-                        path= line.split()[1]
         else:
-            log.error("Cannot find the config files used by the xrootd running process")
+            log.error("Cannot find the config file used by the xrootd running process")
     else:
         log.error("Cannot find a running xrootd process from which obtain the configuration file used")
 
+    return xrootd_config_file
+
+
+# When the configuration parameter 'path' is set to auto, we will try to set it to xrootd's configuration parameter
+# 'oss.localroot'
+def get_localroot_from_xrootd_config_file(xrootd_config_file):
+    # Use xrootd config file to find the 'oss.localroot' parameter
+    out = subprocess.Popen(['cconfig', '-c', xrootd_config_file], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, stderr = out.communicate()
+    if stderr is None:
+        log.debug("Parsing cconfig output to get 'oss.localroot'")
+    else:
+        log.error("Something went wrong when executing cconfig -c "+ xrootd_config_file)
+
+    path=""
+    for line in stdout.split('\n'):
+        if "oss.localroot" in line:
+            if len(line.split()) > 0:
+                path= line.split()[1]
+                # no break intentionally, it could be more than one
+                # we want to take the last one.
     return path
 
+def get_pfc_spaces_from_xrootd_config_file(xrootd_config_file):
+    # 1.  Use xrootd config file to find the 'pfc.spaces' parameter
+    out = subprocess.Popen(['cconfig', '-c', xrootd_config_file], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, stderr = out.communicate()
+    if stderr is None:
+        log.debug("Parsing cconfig output to get 'pfc.spaces'")
+    else:
```

Comment:
This will never happen because you did `stderr=subprocess.STDOUT` above. Also, cconfig prints everything to stderr (try running `cconfig -c /etc/xrootd/xrootd-stash-cache.cfg >/dev/null`).
Try checking `out.returncode != 0` to see if there was an error.

Reply:
I have fixed this with the last commit